### DISCUSSION TITLE
-domain file() What kind of obj files are supported

### DISCUSSION THREAD

**Discussion by Roc-son** [neperfepx/neper#947] [Q&A] [answered]
I want to import the obj file in the -T module, but I get an error, here's a simple example log:
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.10.2-2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 16 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2024, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (/home/roc/.neperrc).
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 10 -id 6 -domain file(test.obj)
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   :     [i] Parsing file test.obj...
Info   :     > Parsing file... 100%
Info   :     > Recovering edges... 100%
Info   :     [i] Parsed file test.obj.
Info   : Creating tessellation...
Segmentation fault (core dumped)

But I've found that the built-in sample house.obj works, so what's the reason? What is the format that neper expects?
Attached here is my obj file:
[test.txt](https://github.com/user-attachments/files/18001386/test.txt)

**Comment by Roc-son** [comment]
In addition, when I try to read the obj file generated by neper himself, I also get an error:
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.10.2-2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 16 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2024, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/roc/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 10 -domain file(n10-id2.obj)
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   :     [i] Parsing file `n10-id2.obj'...
Info   :     > Parsing file... 100%
Info   :     > Recovering edges... 100%
Info   :     > Recovering face geometry... 100%
Info   :     [i] Parsed file `n10-id2.obj'.
Info   : Creating tessellation...
Info   :   - Setting seeds...
Info   :   - Running tessellation...
Info   : Generating crystal orientations...
Info   :   - Setting seeds...
Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.
```

**Reply by rquey** [reply]
The input obj must be a 1-cell tessellation; otherwise, use the `:cell(<cell_id>)` modifier, see [-domain](https://neper.info/doc/neper_t.html#cmdoption-domain).

**Comment by rquey** [comment, marked as answer]
Unfortunately, Neper simply won't work with such a geometry.
![image](https://github.com/user-attachments/assets/12443552-a7c5-4fda-bc0c-a660d68dce5f)
